# Etude de marché du poulet


Faire une analyse pour choisir les pays aptes à l'exportation du poulet pour un lancement à l'international.

Il y a 2 fichiers csv qui sont mis à disposition :

- DisponibiliteAlimentaire_2017.csv : un fichier qui contient les informations sur le types de données récupérées, le pays, à quoi est destiné le produit, le produit, l’année, unité et sa valeur et le symbole et sa description.
- Population_2000_2018.csv : un fichier qui contient les informations sur les types de données, le pays , la population prit en compte, l’année (Ces données sont aussi identifiées par un code), l’unité et sa valeur,  le symbole et sa description et une note.

## Table des matières

* [Import des librairies et des données](#import)  

* [Observation des Dataframes](#dataframe)

* [Rapprochement des exports dispo_alimentaire et population](#merge)


## Import des librairies et des données <a name="import"></a>

In [4]:
import pandas as pd
import numpy as np

dispo_alimentaire = pd.read_csv("C:/Users/Greg/Documents/entreprise/data/Openclassroom/Projet 9 Produisez une étude de marché avec R ou Python/DAN-P9-data/DisponibiliteAlimentaire_2017.csv")
population = pd.read_csv("C:/Users/Greg/Documents/entreprise/data/Openclassroom/Projet 9 Produisez une étude de marché avec R ou Python/DAN-P9-data/Population_2000_2018.csv")

## Observation des Dataframes : <a name="dataframe"></a>  

**Dataframe dispoAlimentaire**  

On va vérifier s'il y a des valeurs manquantes dans le dataframe.

In [5]:
dispo_alimentaire.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 176600 entries, 0 to 176599
Data columns (total 14 columns):
 #   Column                  Non-Null Count   Dtype  
---  ------                  --------------   -----  
 0   Code Domaine            176600 non-null  object 
 1   Domaine                 176600 non-null  object 
 2   Code zone               176600 non-null  int64  
 3   Zone                    176600 non-null  object 
 4   Code Élément            176600 non-null  int64  
 5   Élément                 176600 non-null  object 
 6   Code Produit            176600 non-null  int64  
 7   Produit                 176600 non-null  object 
 8   Code année              176600 non-null  int64  
 9   Année                   176600 non-null  int64  
 10  Unité                   176600 non-null  object 
 11  Valeur                  176600 non-null  float64
 12  Symbole                 176600 non-null  object 
 13  Description du Symbole  176600 non-null  object 
dtypes: float64(1), int64

Le dataframe est composé de 176 600 lignes.  
De plus, on remarque que pour toutes les colonnes, il y a bien aucune valeurs manquantes.

In [6]:
dispo_alimentaire.describe()

Code zone   Code Élément   Code Produit  Code année     Année  \
count  176600.000000  176600.000000  176600.000000    176600.0  176600.0   
mean      125.474740    3810.052894    2628.778545      2017.0    2017.0   
std        72.941699    2211.790089      91.993247         0.0       0.0   
min         1.000000     645.000000    2511.000000      2017.0    2017.0   
25%        63.000000     684.000000    2556.000000      2017.0    2017.0   
50%       120.000000    5142.000000    2612.000000      2017.0    2017.0   
75%       188.000000    5301.000000    2732.000000      2017.0    2017.0   
max       276.000000    5911.000000    2899.000000      2017.0    2017.0   

              Valeur  
count  176600.000000  
mean      210.554918  
std      4762.047787  
min    -10388.000000  
25%         0.000000  
50%         0.120000  
75%         8.462500  
max    758548.000000

Pas de problème pour les valeurs numériques.  
Les données pour ce dataframe concernent bien l'année 2017.  
On affiche le dataframe.

In [7]:
dispo_alimentaire

Code Domaine                      Domaine  Code zone         Zone  \
0               FBS  Nouveaux Bilans Alimentaire          2  Afghanistan   
1               FBS  Nouveaux Bilans Alimentaire          2  Afghanistan   
2               FBS  Nouveaux Bilans Alimentaire          2  Afghanistan   
3               FBS  Nouveaux Bilans Alimentaire          2  Afghanistan   
4               FBS  Nouveaux Bilans Alimentaire          2  Afghanistan   
...             ...                          ...        ...          ...   
176595          FBS  Nouveaux Bilans Alimentaire        181     Zimbabwe   
176596          FBS  Nouveaux Bilans Alimentaire        181     Zimbabwe   
176597          FBS  Nouveaux Bilans Alimentaire        181     Zimbabwe   
176598          FBS  Nouveaux Bilans Alimentaire        181     Zimbabwe   
176599          FBS  Nouveaux Bilans Alimentaire        181     Zimbabwe   

        Code Élément                                            Élément  \
0               5511                                         Production   
1               5611                            Importations - Quantité   
2               5072                                 Variation de stock   
3               5911                            Exportations - Quantité   
4               5301                           Disponibilité intérieure   
...              ...                                                ...   
176595          5142                                         Nourriture   
176596           645  Disponibilité alimentaire en quantité (kg/pers...   
176597           664     Disponibilité alimentaire (Kcal/personne/jour)   
176598           674  Disponibilité de protéines en quantité (g/pers...   
176599           684  Disponibilité de matière grasse en quantité (g...   

        Code Produit          Produit  Code année  Année               Unité  \
0               2511  Blé et produits        2017   2017  Milliers de tonnes   
1               2511  Blé et produits        2017   2017  Milliers de tonnes   
2               2511  Blé et produits        2017   2017  Milliers de tonnes   
3               2511  Blé et produits        2017   2017  Milliers de tonnes   
4               2511  Blé et produits        2017   2017  Milliers de tonnes   
...              ...              ...         ...    ...                 ...   
176595          2899     Miscellanees        2017   2017  Milliers de tonnes   
176596          2899     Miscellanees        2017   2017                  kg   
176597          2899     Miscellanees        2017   2017  Kcal/personne/jour   
176598          2899     Miscellanees        2017   2017     g/personne/jour   
176599          2899     Miscellanees        2017   2017     g/personne/jour   

         Valeur Symbole Description du Symbole  
0       4281.00       S  Données standardisées  
1       2302.00       S  Données standardisées  
2       -119.00       S  Données standardisées  
3          0.00       S  Données standardisées  
4       6701.00       S  Données standardisées  
...         ...     ...                    ...  
176595    19.00       S  Données standardisées  
176596     1.33      Fc        Donnée calculée  
176597     1.00      Fc        Donnée calculée  
176598     0.04      Fc        Donnée calculée  
176599     0.02      Fc        Donnée calculée  

[176600 rows x 14 columns]

On va sélectionner les informations utiles pour l'analyse, en commencent par la colonne produit, on va choisir les données qui contiennent du poulet.  
Puis on va supprimer les données non utiles.

In [8]:
dispo_alimentaire['Produit'].unique()

array(['Blé et produits', 'Riz et produits', 'Orge et produits',
       'Maïs et produits', 'Seigle et produits', 'Avoine',
       'Millet et produits', 'Sorgho et produits', 'Céréales, Autres',
       'Pommes de Terre et produits', 'Ignames', 'Racines nda',
       'Sucre, canne', 'Sucre, betterave', 'Sucre Eq Brut',
       'Edulcorants Autres', 'Miel', 'Haricots', 'Pois',
       'Légumineuses Autres et produits', 'Noix et produits', 'Soja',
       'Arachides Decortiquees', 'Graines de tournesol',
       'Graines Colza/Moutarde', 'Graines de coton', 'Coco (Incl Coprah)',
       'Sésame', 'Olives', 'Plantes Oleiferes, Autre', 'Huile de Soja',
       "Huile d'Arachide", 'Huile de Tournesol',
       'Huile de Colza&Moutarde', 'Huile Graines de Coton',
       'Huile de Palmistes', 'Huile de Palme', 'Huile de Coco',
       'Huile de Sésame', "Huile d'Olive", 'Huile de Son de Riz',
       'Huile de Germe de Maïs', 'Huil Plantes Oleif Autr',
       'Tomates et produits', 'Oignons', 'Légumes, 

En affichant, les données contenues dans la colonne produit, la donnée qui nous intérésse ici est la viande de volailles.  
On va filtrer le dataframe avec cette donnée.

In [9]:
dispo_alimentaire = dispo_alimentaire.loc[dispo_alimentaire['Produit'] == 'Viande de Volailles', :]

dispo_alimentaire

Code Domaine                      Domaine  Code zone         Zone  \
651             FBS  Nouveaux Bilans Alimentaire          2  Afghanistan   
652             FBS  Nouveaux Bilans Alimentaire          2  Afghanistan   
653             FBS  Nouveaux Bilans Alimentaire          2  Afghanistan   
654             FBS  Nouveaux Bilans Alimentaire          2  Afghanistan   
655             FBS  Nouveaux Bilans Alimentaire          2  Afghanistan   
...             ...                          ...        ...          ...   
176396          FBS  Nouveaux Bilans Alimentaire        181     Zimbabwe   
176397          FBS  Nouveaux Bilans Alimentaire        181     Zimbabwe   
176398          FBS  Nouveaux Bilans Alimentaire        181     Zimbabwe   
176399          FBS  Nouveaux Bilans Alimentaire        181     Zimbabwe   
176400          FBS  Nouveaux Bilans Alimentaire        181     Zimbabwe   

        Code Élément                                            Élément  \
651             5511                                         Production   
652             5611                            Importations - Quantité   
653             5072                                 Variation de stock   
654             5301                           Disponibilité intérieure   
655             5123                                             Pertes   
...              ...                                                ...   
176396          5142                                         Nourriture   
176397           645  Disponibilité alimentaire en quantité (kg/pers...   
176398           664     Disponibilité alimentaire (Kcal/personne/jour)   
176399           674  Disponibilité de protéines en quantité (g/pers...   
176400           684  Disponibilité de matière grasse en quantité (g...   

        Code Produit              Produit  Code année  Année  \
651             2734  Viande de Volailles        2017   2017   
652             2734  Viande de Volailles        2017   2017   
653             2734  Viande de Volailles        2017   2017   
654             2734  Viande de Volailles        2017   2017   
655             2734  Viande de Volailles        2017   2017   
...              ...                  ...         ...    ...   
176396          2734  Viande de Volailles        2017   2017   
176397          2734  Viande de Volailles        2017   2017   
176398          2734  Viande de Volailles        2017   2017   
176399          2734  Viande de Volailles        2017   2017   
176400          2734  Viande de Volailles        2017   2017   

                     Unité  Valeur Symbole Description du Symbole  
651     Milliers de tonnes   28.00       S  Données standardisées  
652     Milliers de tonnes   29.00       S  Données standardisées  
653     Milliers de tonnes    0.00       S  Données standardisées  
654     Milliers de tonnes   57.00       S  Données standardisées  
655     Milliers de tonnes    2.00       S  Données standardisées  
...                    ...     ...     ...                    ...  
176396  Milliers de tonnes   67.00       S  Données standardisées  
176397                  kg    4.68      Fc        Donnée calculée  
176398  Kcal/personne/jour   16.00      Fc        Donnée calculée  
176399     g/personne/jour    1.59      Fc        Donnée calculée  
176400     g/personne/jour    0.99      Fc        Donnée calculée  

[2061 rows x 14 columns]

On passe à la suppression des données.    
Les colonnes qui contiennent le code vont être supprimées.  
De plus, les colonnes domaine, produit, année, unité, symbole et description du symbole vont aussi être supprimées.

In [10]:
dispo_alimentaire = dispo_alimentaire.drop(columns =['Code Domaine', 'Domaine', 'Code zone', 'Code Élément', 'Code Produit', 'Produit', 'Code année', 'Année', 'Unité', 'Symbole', 'Description du Symbole'])

dispo_alimentaire

Zone                                            Élément  Valeur
651     Afghanistan                                         Production   28.00
652     Afghanistan                            Importations - Quantité   29.00
653     Afghanistan                                 Variation de stock    0.00
654     Afghanistan                           Disponibilité intérieure   57.00
655     Afghanistan                                             Pertes    2.00
...             ...                                                ...     ...
176396     Zimbabwe                                         Nourriture   67.00
176397     Zimbabwe  Disponibilité alimentaire en quantité (kg/pers...    4.68
176398     Zimbabwe     Disponibilité alimentaire (Kcal/personne/jour)   16.00
176399     Zimbabwe  Disponibilité de protéines en quantité (g/pers...    1.59
176400     Zimbabwe  Disponibilité de matière grasse en quantité (g...    0.99

[2061 rows x 3 columns]

On va afficher le contenu des données de la colonne élément qui semble avoir une importance pour l'analyse.

In [11]:
dispo_alimentaire['Élément'].unique()

array(['Production', 'Importations - Quantité', 'Variation de stock',
       'Disponibilité intérieure', 'Pertes', 'Résidus', 'Nourriture',
       'Disponibilité alimentaire en quantité (kg/personne/an)',
       'Disponibilité alimentaire (Kcal/personne/jour)',
       'Disponibilité de protéines en quantité (g/personne/jour)',
       'Disponibilité de matière grasse en quantité (g/personne/jour)',
       'Exportations - Quantité', 'Alimentation pour touristes',
       'Traitement', 'Autres utilisations (non alimentaire)',
       'Aliments pour animaux', 'Semences'], dtype=object)

Les informations sont importantes pour l'analyse.  
On va les afficher en colonne plutôt qu'en ligne.

In [12]:
# On effectue un pivot_table sur la colonne élément.
dispo_alimentaire = dispo_alimentaire.pivot_table(values='Valeur', index='Zone', columns='Élément', aggfunc=np.sum).reset_index()

dispo_alimentaire

Élément                   Zone  Alimentation pour touristes  \
0                  Afghanistan                          NaN   
1               Afrique du Sud                          0.0   
2                      Albanie                          NaN   
3                      Algérie                          0.0   
4                    Allemagne                          NaN   
..                         ...                          ...   
167        Émirats arabes unis                          NaN   
168                   Équateur                          0.0   
169      États-Unis d'Amérique                          NaN   
170                   Éthiopie                          0.0   
171               Îles Salomon                          0.0   

Élément  Aliments pour animaux  Autres utilisations (non alimentaire)  \
0                          NaN                                    NaN   
1                          NaN                                    NaN   
2                          NaN                                    NaN   
3                          NaN                                    NaN   
4                          NaN                                    NaN   
..                         ...                                    ...   
167                        NaN                                    NaN   
168                        NaN                                    NaN   
169                        NaN                                   89.0   
170                        NaN                                    NaN   
171                        NaN                                    NaN   

Élément  Disponibilité alimentaire (Kcal/personne/jour)  \
0                                                   5.0   
1                                                 143.0   
2                                                  85.0   
3                                                  22.0   
4                                                  71.0   
..                                                  ...   
167                                               147.0   
168                                                83.0   
169                                               219.0   
170                                                 0.0   
171                                                18.0   

Élément  Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                                     1.53        
1                                                    35.69        
2                                                    16.36        
3                                                     6.38        
4                                                    19.47        
..                                                     ...        
167                                                  43.47        
168                                                  19.31        
169                                                  55.68        
170                                                   0.13        
171                                                   4.45        

Élément  Disponibilité de matière grasse en quantité (g/personne/jour)  \
0                                                     0.33               
1                                                     9.25               
2                                                     6.45               
3                                                     1.50               
4                                                     4.16               
..                                                     ...               
167                                                   9.25               
168                                                   6.35               
169                                                  14.83               
170                                                   0.03               
171                                   

Après avoir pivoté, on va faire le tri dans les colonnes pour garder celle utile pour l'analyse.  
Les colonnes utiles sont : 
- Disponibilité alimentaire (Kcal/personne/jour).
- Disponibilité de protéines en quantité (g/personne/jour).
- Exportations - Quantité.
- Importations - Quantité.
- Production.

In [13]:
dispo_alimentaire = dispo_alimentaire.drop(columns =['Alimentation pour touristes', 'Aliments pour animaux', 'Autres utilisations (non alimentaire)', 'Disponibilité alimentaire en quantité (kg/personne/an)', 'Disponibilité de matière grasse en quantité (g/personne/jour)', 'Disponibilité intérieure', 'Nourriture', 'Pertes', 'Résidus', 'Semences', 'Traitement', 'Variation de stock'])

dispo_alimentaire

Élément                   Zone  \
0                  Afghanistan   
1               Afrique du Sud   
2                      Albanie   
3                      Algérie   
4                    Allemagne   
..                         ...   
167        Émirats arabes unis   
168                   Équateur   
169      États-Unis d'Amérique   
170                   Éthiopie   
171               Îles Salomon   

Élément  Disponibilité alimentaire (Kcal/personne/jour)  \
0                                                   5.0   
1                                                 143.0   
2                                                  85.0   
3                                                  22.0   
4                                                  71.0   
..                                                  ...   
167                                               147.0   
168                                                83.0   
169                                               219.0   
170                                                 0.0   
171                                                18.0   

Élément  Disponibilité de protéines en quantité (g/personne/jour)  \
0                                                     0.54          
1                                                    14.11          
2                                                     6.26          
3                                                     1.97          
4                                                     7.96          
..                                                     ...          
167                                                  14.80          
168                                                   6.15          
169                                                  19.93          
170                                                   0.04          
171                                                   1.51          

Élément  Exportations - Quantité  Importations - Quantité  Production  
0                            NaN                     29.0        28.0  
1                           63.0                    514.0      1667.0  
2                            0.0                     38.0        13.0  
3                            0.0                      2.0       275.0  
4                          646.0                    842.0      1514.0  
..                           ...                      ...         ...  
167                         94.0                    433.0        48.0  
168                          0.0                      0.0       340.0  
169                       3692.0                    123.0     21914.0  
170                          NaN                      1.0        14.0  
171                          0.0                      6.0         0.0  

[172 rows x 6 columns]

On renomme ensuite la colonne zone en pays.

In [14]:
dispo_alimentaire = dispo_alimentaire.rename(columns={'Zone' : 'Pays'})

dispo_alimentaire.head()

Élément            Pays  Disponibilité alimentaire (Kcal/personne/jour)  \
0           Afghanistan                                             5.0   
1        Afrique du Sud                                           143.0   
2               Albanie                                            85.0   
3               Algérie                                            22.0   
4             Allemagne                                            71.0   

Élément  Disponibilité de protéines en quantité (g/personne/jour)  \
0                                                     0.54          
1                                                    14.11          
2                                                     6.26          
3                                                     1.97          
4                                                     7.96          

Élément  Exportations - Quantité  Importations - Quantité  Production  
0                            NaN                     29.0        28.0  
1                           63.0                    514.0      1667.0  
2                            0.0                     38.0        13.0  
3                            0.0                      2.0       275.0  
4                          646.0                    842.0      1514.0

On contrôle s'il ny a pas de doublon dans la colonne pays car elle va nous servir pour la fusion avec le dataframe population.

In [15]:
def doublons(df): # Nombre de doublons dans la dataframe
    print(len(df) - len(df.drop_duplicates()), 'doublons')

doublons(dispo_alimentaire['Pays'])

0 doublons


On passe à l'analyse du dataframe population.

**Dataframe population**  

On regarde les valeurs manquantes.

In [16]:
population.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4411 entries, 0 to 4410
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Code Domaine            4411 non-null   object 
 1   Domaine                 4411 non-null   object 
 2   Code zone               4411 non-null   int64  
 3   Zone                    4411 non-null   object 
 4   Code Élément            4411 non-null   int64  
 5   Élément                 4411 non-null   object 
 6   Code Produit            4411 non-null   int64  
 7   Produit                 4411 non-null   object 
 8   Code année              4411 non-null   int64  
 9   Année                   4411 non-null   int64  
 10  Unité                   4411 non-null   object 
 11  Valeur                  4411 non-null   float64
 12  Symbole                 4411 non-null   object 
 13  Description du Symbole  4411 non-null   object 
 14  Note                    258 non-null    

Le dataframe est composé de 4 411 lignes.  
Il n'y a pas de données manquantes sauf pour la colonne note mais ce n'est pas important car c'est une colonne non utile pour l'analyse.

In [17]:
population.describe()

Code zone  Code Élément  Code Produit   Code année        Année  \
count  4411.000000        4411.0        4411.0  4411.000000  4411.000000   
mean    132.202902         511.0        3010.0  2009.068238  2009.068238   
std      75.854840           0.0           0.0     5.481539     5.481539   
min       1.000000         511.0        3010.0  2000.000000  2000.000000   
25%      68.000000         511.0        3010.0  2004.000000  2004.000000   
50%     132.000000         511.0        3010.0  2009.000000  2009.000000   
75%     195.000000         511.0        3010.0  2014.000000  2014.000000   
max     299.000000         511.0        3010.0  2018.000000  2018.000000   

             Valeur  
count  4.411000e+03  
mean   2.963004e+04  
std    1.238029e+05  
min    7.850000e-01  
25%    3.921890e+02  
50%    4.764741e+03  
75%    1.821548e+04  
max    1.427648e+06

Pas de problème au niveau des données.  
L'année va de 2000 à 2018, par la suite, pour garder les informations sur une même année, on va choisir l'année 2017 car le dataframe dispo_alimentaire concerne la période de 2017.

In [18]:
population

Code Domaine                       Domaine  Code zone         Zone  \
0              OA  Séries temporelles annuelles          2  Afghanistan   
1              OA  Séries temporelles annuelles          2  Afghanistan   
2              OA  Séries temporelles annuelles          2  Afghanistan   
3              OA  Séries temporelles annuelles          2  Afghanistan   
4              OA  Séries temporelles annuelles          2  Afghanistan   
...           ...                           ...        ...          ...   
4406           OA  Séries temporelles annuelles        181     Zimbabwe   
4407           OA  Séries temporelles annuelles        181     Zimbabwe   
4408           OA  Séries temporelles annuelles        181     Zimbabwe   
4409           OA  Séries temporelles annuelles        181     Zimbabwe   
4410           OA  Séries temporelles annuelles        181     Zimbabwe   

      Code Élément            Élément  Code Produit                 Produit  \
0              511  Population totale          3010  Population-Estimations   
1              511  Population totale          3010  Population-Estimations   
2              511  Population totale          3010  Population-Estimations   
3              511  Population totale          3010  Population-Estimations   
4              511  Population totale          3010  Population-Estimations   
...            ...                ...           ...                     ...   
4406           511  Population totale          3010  Population-Estimations   
4407           511  Population totale          3010  Population-Estimations   
4408           511  Population totale          3010  Population-Estimations   
4409           511  Population totale          3010  Population-Estimations   
4410           511  Population totale          3010  Population-Estimations   

      Code année  Année           Unité     Valeur Symbole  \
0           2000   2000  1000 personnes  20779.953       X   
1           2001   2001  1000 personnes  21606.988       X   
2           2002   2002  1000 personnes  22600.770       X   
3           2003   2003  1000 personnes  23680.871       X   
4           2004   2004  1000 personnes  24726.684       X   
...          ...    ...             ...        ...     ...   
4406        2014   2014  1000 personnes  13586.707       X   
4407        2015   2015  1000 personnes  13814.629       X   
4408        2016   2016  1000 personnes  14030.331       X   
4409        2017   2017  1000 personnes  14236.595       X   
4410        2018   2018  1000 personnes  14438.802       X   

             Description du Symbole Note  
0     Sources internationales sûres  NaN  
1     Sources internationales sûres  NaN  
2     Sources internationales sûres  NaN  
3     Sources internationales sûres  NaN  
4     Sources internationales sûres  NaN  
...                             ...  ...  
4406  Sources internationales sûres  NaN  
4407  Sources internationales sûres  NaN  
4408  Sources internationales sûres  NaN  
4409  Sources internationales sûres  NaN  
4410  Sources internationales sûres  NaN  

[4411 rows x 15 columns]

On va d'abord filtrer par l'année 2017.  
Puis comme pour le dataframe dispo_alimentaire, on va supprimer les mêmes colonnes et en plus les colonnes élément et note.

In [19]:
population = population.loc[population['Année'] == 2017, :]

population

Code Domaine                       Domaine  Code zone  \
17             OA  Séries temporelles annuelles          2   
36             OA  Séries temporelles annuelles        202   
55             OA  Séries temporelles annuelles          3   
74             OA  Séries temporelles annuelles          4   
93             OA  Séries temporelles annuelles         79   
...           ...                           ...        ...   
4333           OA  Séries temporelles annuelles        236   
4352           OA  Séries temporelles annuelles        237   
4371           OA  Séries temporelles annuelles        249   
4390           OA  Séries temporelles annuelles        251   
4409           OA  Séries temporelles annuelles        181   

                                        Zone  Code Élément            Élément  \
17                               Afghanistan           511  Population totale   
36                            Afrique du Sud           511  Population totale   
55                                   Albanie           511  Population totale   
74                                   Algérie           511  Population totale   
93                                 Allemagne           511  Population totale   
...                                      ...           ...                ...   
4333  Venezuela (République bolivarienne du)           511  Population totale   
4352                                Viet Nam           511  Population totale   
4371                                   Yémen           511  Population totale   
4390                                  Zambie           511  Population totale   
4409                                Zimbabwe           511  Population totale   

      Code Produit                 Produit  Code année  Année           Unité  \
17            3010  Population-Estimations        2017   2017  1000 personnes   
36            3010  Population-Estimations        2017   2017  1000 personnes   
55            3010  Population-Estimations        2017   2017  1000 personnes   
74            3010  Population-Estimations        2017   2017  1000 personnes   
93            3010  Population-Estimations        2017   2017  1000 personnes   
...            ...                     ...         ...    ...             ...   
4333          3010  Population-Estimations        2017   2017  1000 personnes   
4352          3010  Population-Estimations        2017   2017  1000 personnes   
4371          3010  Population-Estimations        2017   2017  1000 personnes   
4390          3010  Population-Estimations        2017   2017  1000 personnes   
4409          3010  Population-Estimations        2017   2017  1000 personnes   

         Valeur Symbole         Description du Symbole Note  
17    36296.113       X  Sources internationales sûres  NaN  
36    57009.756       X  Sources internationales sûres  NaN  
55     2884.169       X  Sources internationales sûres  NaN  
74    41389.189       X  Sources internationales sûres  NaN  
93    82658.409       X  Sources internationales sûres  NaN  
...         ...     ...                            ...  ...  
4333  29402.484       X  Sources internationales sûres  NaN  
4352  94600.648       X  Sources internationales sûres  NaN  
4371  27834.819       X  Sources internationales sûres  NaN  
4390  16853.599       X  Sources internationales sûres  NaN  
4409  14236.595       X  Sources internationales sûres  NaN  

[236 rows x 15 columns]

On passe ensuite à la suppression.

In [20]:
population = population.drop(columns =['Code Domaine', 'Domaine', 'Code zone', 'Code Élément', 'Élément', 'Code Produit', 'Produit', 'Code année','Année', 'Unité', 'Symbole', 'Description du Symbole', 'Note'])

population

Zone     Valeur
17                               Afghanistan  36296.113
36                            Afrique du Sud  57009.756
55                                   Albanie   2884.169
74                                   Algérie  41389.189
93                                 Allemagne  82658.409
...                                      ...        ...
4333  Venezuela (République bolivarienne du)  29402.484
4352                                Viet Nam  94600.648
4371                                   Yémen  27834.819
4390                                  Zambie  16853.599
4409                                Zimbabwe  14236.595

[236 rows x 2 columns]

On renomme les colonnes.

In [21]:
population = population.rename(columns={'Zone' : 'Pays', 'Valeur' : 'Population pour 1000'})

population

Pays  Population pour 1000
17                               Afghanistan             36296.113
36                            Afrique du Sud             57009.756
55                                   Albanie              2884.169
74                                   Algérie             41389.189
93                                 Allemagne             82658.409
...                                      ...                   ...
4333  Venezuela (République bolivarienne du)             29402.484
4352                                Viet Nam             94600.648
4371                                   Yémen             27834.819
4390                                  Zambie             16853.599
4409                                Zimbabwe             14236.595

[236 rows x 2 columns]

On vérifie s'il n'y a pas de doublon.

In [22]:
doublons(population['Pays'])

0 doublons


On va passer à la partie fusion des dataframes.

## Rapprochement des exports dispo_alimentaire et population <a name="merge"></a>

In [23]:
# On merge (inner) les 2 dataframes.
etude_poulet = dispo_alimentaire.merge(population, on='Pays')

etude_poulet

Pays  Disponibilité alimentaire (Kcal/personne/jour)  \
0              Afghanistan                                             5.0   
1           Afrique du Sud                                           143.0   
2                  Albanie                                            85.0   
3                  Algérie                                            22.0   
4                Allemagne                                            71.0   
..                     ...                                             ...   
167    Émirats arabes unis                                           147.0   
168               Équateur                                            83.0   
169  États-Unis d'Amérique                                           219.0   
170               Éthiopie                                             0.0   
171           Îles Salomon                                            18.0   

     Disponibilité de protéines en quantité (g/personne/jour)  \
0                                                 0.54          
1                                                14.11          
2                                                 6.26          
3                                                 1.97          
4                                                 7.96          
..                                                 ...          
167                                              14.80          
168                                               6.15          
169                                              19.93          
170                                               0.04          
171                                               1.51          

     Exportations - Quantité  Importations - Quantité  Production  \
0                        NaN                     29.0        28.0   
1                       63.0                    514.0      1667.0   
2                        0.0                     38.0        13.0   
3                        0.0                      2.0       275.0   
4                      646.0                    842.0      1514.0   
..                       ...                      ...         ...   
167                     94.0                    433.0        48.0   
168                      0.0                      0.0       340.0   
169                   3692.0                    123.0     21914.0   
170                      NaN                      1.0        14.0   
171                      0.0                      6.0         0.0   

     Population pour 1000  
0               36296.113  
1               57009.756  
2                2884.169  
3               41389.189  
4               82658.409  
..                    ...  
167              9487.203  
168             16785.361  
169            325084.756  
170            106399.924  
171               636.039  

[172 rows x 7 columns]

Pas de problème avec le merge, on a bien les 172 lignes.  
On a fait le choix d'un merge en inner car on veut seulement les données qui correspondent aux 2 dataframes.  

On enlève dans les pays, la France car l'entreprise est implantée en France et que notre étude concerne l'implantation à l'international.  
De plus , la Chine, les Etats-Unis, l'Inde et le Brésil vont être supprimés car ce sont des pays qui une forte production de poulet et de population. Il n'est pas pertinent de les viser pour une implantation et en plus, leurs données ne sont pas pertinentes pour l'analyse.

In [24]:
etude_poulet.drop(etude_poulet.loc[(etude_poulet['Pays'] == 'France') | (etude_poulet['Pays'] == 'Inde') | (etude_poulet['Pays'] == 'Brésil') | (etude_poulet['Pays'] == "États-Unis d'Amérique") | (etude_poulet['Pays'] == 'Chine, continentale')].index, inplace=True)

In [25]:
etude_poulet.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 167 entries, 0 to 171
Data columns (total 7 columns):
 #   Column                                                    Non-Null Count  Dtype  
---  ------                                                    --------------  -----  
 0   Pays                                                      167 non-null    object 
 1   Disponibilité alimentaire (Kcal/personne/jour)            167 non-null    float64
 2   Disponibilité de protéines en quantité (g/personne/jour)  167 non-null    float64
 3   Exportations - Quantité                                   130 non-null    float64
 4   Importations - Quantité                                   165 non-null    float64
 5   Production                                                163 non-null    float64
 6   Population pour 1000                                      167 non-null    float64
dtypes: float64(6), object(1)
memory usage: 10.4+ KB


Les valeurs ont bien été supprimées.  
Il y a la présence de valeurs nulles, on va utiliser la moyenne pour les compléter.

In [26]:
# Récupération de la moyenne des exportations.
moy_export = round(etude_poulet['Exportations - Quantité'].mean(),2)

# Récupération de la moyenne des importations.
moy_import = round(etude_poulet['Importations - Quantité'].mean(),2)

# Récupération de la moyenne de la production.
moy_produc = round(etude_poulet['Production'].mean(),2)

In [27]:
# Ajout des moyennes.
etude_poulet['Exportations - Quantité'] = etude_poulet['Exportations - Quantité'].fillna(moy_export)
etude_poulet['Importations - Quantité'] = etude_poulet['Importations - Quantité'].fillna(moy_import)
etude_poulet['Production'] = etude_poulet['Production'].fillna(moy_produc)

In [28]:
etude_poulet.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 167 entries, 0 to 171
Data columns (total 7 columns):
 #   Column                                                    Non-Null Count  Dtype  
---  ------                                                    --------------  -----  
 0   Pays                                                      167 non-null    object 
 1   Disponibilité alimentaire (Kcal/personne/jour)            167 non-null    float64
 2   Disponibilité de protéines en quantité (g/personne/jour)  167 non-null    float64
 3   Exportations - Quantité                                   167 non-null    float64
 4   Importations - Quantité                                   167 non-null    float64
 5   Production                                                167 non-null    float64
 6   Population pour 1000                                      167 non-null    float64
dtypes: float64(6), object(1)
memory usage: 10.4+ KB


Il n'y a plus de valeurs nulles, on peut passer à l'analyse.

In [30]:
etude_poulet

Pays  Disponibilité alimentaire (Kcal/personne/jour)  \
0            Afghanistan                                             5.0   
1         Afrique du Sud                                           143.0   
2                Albanie                                            85.0   
3                Algérie                                            22.0   
4              Allemagne                                            71.0   
..                   ...                                             ...   
166               Égypte                                            50.0   
167  Émirats arabes unis                                           147.0   
168             Équateur                                            83.0   
170             Éthiopie                                             0.0   
171         Îles Salomon                                            18.0   

     Disponibilité de protéines en quantité (g/personne/jour)  \
0                                                 0.54          
1                                                14.11          
2                                                 6.26          
3                                                 1.97          
4                                                 7.96          
..                                                 ...          
166                                               4.51          
167                                              14.80          
168                                               6.15          
170                                               0.04          
171                                               1.51          

     Exportations - Quantité  Importations - Quantité  Production  \
0                      68.07                     29.0        28.0   
1                      63.00                    514.0      1667.0   
2                       0.00                     38.0        13.0   
3                       0.00                      2.0       275.0   
4                     646.00                    842.0      1514.0   
..                       ...                      ...         ...   
166                     1.00                    110.0      1118.0   
167                    94.00                    433.0        48.0   
168                     0.00                      0.0       340.0   
170                    68.07                      1.0        14.0   
171                     0.00                      6.0         0.0   

     Population pour 1000  
0               36296.113  
1               57009.756  
2                2884.169  
3               41389.189  
4               82658.409  
..                    ...  
166             96442.591  
167              9487.203  
168             16785.361  
170            106399.924  
171               636.039  

[167 rows x 7 columns]

In [31]:
etude_poulet.to_csv("C:/Users/Greg/Documents/entreprise/data/Openclassroom/Projet 9 Produisez une étude de marché avec R ou Python/export/Etude poulet.csv", index = False)